src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -198,7 +239,7 @@ static ucs_status_t uct_cuda_ipc_iface_query(uct_iface_h tl_iface,
 
     uct_base_iface_query(&iface->super, iface_attr);
 
-    iface_attr->iface_addr_len          = sizeof(pid_t);
+    iface_attr->iface_addr_len          = sizeof(uct_cuda_base_sys_dev_map_t);
```

Comment:
can we define the length according to the real number of gpus? (to avoid unnecessary address increase)

Reply:
@brminich Kept it this way to handle the case when different processes within the same node use different number of devices in CUDA_VISIBLE_DEVICES. For example, if process 0 uses CUDA_VISIBLE_DEVICES=0 and process 1 uses CUDA_VISIBLE_DEVICES=1,2 then the instances of cuda_ipc iface for the two processes will be different. I wasn't sure if iface addresses have to be consistent across processes during iface_addr exchange.

Reply:
i think it is ok to have different address lengths, but it will not work with unified mode (which is disabled by default)
@yosefe, wdyt?